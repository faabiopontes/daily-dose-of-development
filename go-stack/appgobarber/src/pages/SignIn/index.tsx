import React, { useCallback, useRef } from 'react';
import { Image, KeyboardAvoidingView, Platform, View } from 'react-native';

import Icon from 'react-native-vector-icons/Feather';
import { ScrollView } from 'react-native-gesture-handler';
import { useNavigation } from '@react-navigation/native';
import { Form } from '@unform/mobile';
import { FormHandles } from '@unform/core';
import {
  Container,
  Title,
  ForgotPassword,
  ForgotPasswordText,
  CreateAccountButton,
  CreateAccountButtonText,
} from './styles';
import logoImg from '../../assets/logo.png';
import Input from '../../components/Input';
import Button from '../../components/Button';

interface ISignInFormInputs {
  email: string;
  password: string;
}

const SignIn: React.FC = () => {
  const navigation = useNavigation();
  const formRef = useRef<FormHandles>(null);

  const handleSignIn = useCallback((data: ISignInFormInputs) => {
    console.log(data);
  }, []);

  return (
    <>
      <KeyboardAvoidingView
        style={{ flex: 1 }}
        behavior={Platform.OS === 'ios' ? 'padding' : undefined}
      >
        <ScrollView
          keyboardShouldPersistTaps="handled"
          contentContainerStyle={{ flex: 1 }}
        >
          <Container>
            <Image source={logoImg} />

            <View>
              <Title>Fa√ßa seu logon</Title>
            </View>

            <Form ref={formRef} onSubmit={handleSignIn}>
              <Input name="email" icon="mail" placeholder="E-mail" />

              <Input name="password" icon="lock" placeholder="Senha" />
            </Form>

            <Button onPress={() => formRef.current?.submitForm()}>
              Entrar
            </Button>

            <ForgotPassword
              onPress={() => {
                console.log('Esqueci minha senha');
              }}
            >
              <ForgotPasswordText>Esqueci minha senha</ForgotPasswordText>
            </ForgotPassword>
          </Container>
        </ScrollView>
      </KeyboardAvoidingView>

      <CreateAccountButton onPress={() => navigation.navigate('SignUp')}>
        <Icon name="log-in" size={20} color="#ff9000" />
        <CreateAccountButtonText>Criar uma conta</CreateAccountButtonText>
      </CreateAccountButton>
    </>
  );
};

export default SignIn;
