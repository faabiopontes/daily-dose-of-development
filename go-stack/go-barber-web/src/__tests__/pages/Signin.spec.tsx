import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import SignIn from '../../pages/Signin';

const mockedHistoryPush = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    Link: ({ children }: { children: React.ReactNode }) => children,
    useHistory: () => ({
      push: mockedHistoryPush,
    }),
  };
});

jest.mock('../../hooks/auth', () => ({
  useAuth: () => ({
    signIn: jest.fn(),
  }),
}));

jest.mock('../../hooks/toast', () => ({
  useToast: () => ({
    addToast: jest.fn(),
  }),
}));

describe('SignIn Page', () => {
  it('should be able to sign in', async () => {
    const { getByPlaceholderText, getByText } = render(<SignIn />);

    const emailField = getByPlaceholderText('E-mail');
    const passwordField = getByPlaceholderText('Senha');
    const buttonElement = getByText('Entrar');

    fireEvent.change(emailField, {
      target: { value: 'faabiopontes@gmail.com' },
    });
    fireEvent.change(passwordField, {
      target: { value: '12345678' },
    });

    fireEvent.click(buttonElement);

    await waitFor(() => expect(mockedHistoryPush).toHaveBeenCalledWith('/dashboard'));
  });

  it('should be rendered', () => {
    const component = render(<SignIn />);
    expect(component).toMatchSnapshot();
  });
});
