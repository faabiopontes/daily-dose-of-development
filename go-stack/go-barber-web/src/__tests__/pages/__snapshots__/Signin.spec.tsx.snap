// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Page should be rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ ffkxDX"
      >
        <div
          class="sc-gsTCUz dDqAGx"
        >
          <div
            class="sc-dlfnbm bRhDTv"
          >
            <img
              alt="App Logo"
              src="logo.svg"
            />
            <form>
              <h1>
                Faça seu logon
              </h1>
              <div
                class="sc-jSgupP gXzXci"
              >
                <svg
                  fill="none"
                  height="20"
                  size="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
                  />
                  <polyline
                    points="22,6 12,13 2,6"
                  />
                </svg>
                <input
                  placeholder="E-mail"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="sc-jSgupP gXzXci"
              >
                <svg
                  fill="none"
                  height="20"
                  size="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="11"
                    rx="2"
                    ry="2"
                    width="18"
                    x="3"
                    y="11"
                  />
                  <path
                    d="M7 11V7a5 5 0 0 1 10 0v4"
                  />
                </svg>
                <input
                  placeholder="Senha"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="sc-iBPRYJ eBrNA-d"
                type="submit"
              >
                Entrar
              </button>
              <a
                href="forgot"
              >
                Esqueci minha senha
              </a>
            </form>
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
              />
              <polyline
                points="10 17 15 12 10 7"
              />
              <line
                x1="15"
                x2="3"
                y1="12"
                y2="12"
              />
            </svg>
            Criar conta
          </div>
        </div>
        <div
          class="sc-hKgILt cxShHz"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ ffkxDX"
    >
      <div
        class="sc-gsTCUz dDqAGx"
      >
        <div
          class="sc-dlfnbm bRhDTv"
        >
          <img
            alt="App Logo"
            src="logo.svg"
          />
          <form>
            <h1>
              Faça seu logon
            </h1>
            <div
              class="sc-jSgupP gXzXci"
            >
              <svg
                fill="none"
                height="20"
                size="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
                />
                <polyline
                  points="22,6 12,13 2,6"
                />
              </svg>
              <input
                placeholder="E-mail"
                type="text"
                value=""
              />
            </div>
            <div
              class="sc-jSgupP gXzXci"
            >
              <svg
                fill="none"
                height="20"
                size="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="11"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="11"
                />
                <path
                  d="M7 11V7a5 5 0 0 1 10 0v4"
                />
              </svg>
              <input
                placeholder="Senha"
                type="password"
                value=""
              />
            </div>
            <button
              class="sc-iBPRYJ eBrNA-d"
              type="submit"
            >
              Entrar
            </button>
            <a
              href="forgot"
            >
              Esqueci minha senha
            </a>
          </form>
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
            />
            <polyline
              points="10 17 15 12 10 7"
            />
            <line
              x1="15"
              x2="3"
              y1="12"
              y2="12"
            />
          </svg>
          Criar conta
        </div>
      </div>
      <div
        class="sc-hKgILt cxShHz"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
